tests/test_threadlocal.py

```diff
@@ -0,0 +1,87 @@
+# All Rights Reserved.
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+"""Tests for threadlocal dict module."""
+
+import random
+import string
+import threading
+import Queue
+import unittest
+
+from simpl import threadlocal
+
+
+class TestThreadLocal(unittest.TestCase):
+
+    def setUp(self):
+        # clear the threadlocal dict
+        threadlocal.get_context().clear()
+
+    def get_some_text(self, length=16):
+
+        return ''.join(
+            [random.choice(string.ascii_letters) for _ in xrange(length)])
+
+    def test_get_context(self):
+
+        self.assertIs(threadlocal.get_context(), threadlocal.CONTEXT)
+
+    def test_threadlocal_dict_repr(self):
+
+        tld = threadlocal.get_context()
+        tld['good'] = 'glob'
+        repr_str = repr(tld)
+        expected = "<ThreadLocalDict {'good': 'glob'}>"
+        self.assertEqual(repr_str, expected)
+
+    def test_correct_threadlocal_dicts(self):
+
+        key = 'value'
+        num_threads = 5
+
+        def get_tld(container):
+            current = threadlocal.get_context()
+            current[key] = self.get_some_text()
+            container.put(current._get_local_dict())
+
+        tcon = Queue.Queue()
+        joins = []
```

Comment:
IMO, this code would be a bit easier to understand when first read if these variables were `local_dicts` and `threads`.

```python
local_dicts = Queue.queue()
threads = []
```